ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/CommandDeduplication.scala

```diff
@@ -3,90 +3,141 @@
+            // This allows us to compare the record times directly
+            recordTime.toInstant
+          case None =>
+            // We select minimum record time for pre-execution
+            // During pre-execution in the deduplication state value we stored the maximum record time
+            // To guarantee the deduplication duration, we basically compare the maximum record time of the previous transaction
+            // with the minimum record time of the current transaction. This gives us the smallest possible interval between two transactions.
+            commitContext.minimumRecordTime
+              .getOrElse(
+                throw Err.InternalError("Minimum record time is not set for pre-execution")
+              )
+              .toInstant
+        }
+        val isDuplicate = maybeDedupValue
+          .flatMap(commandDeduplication =>
+            commandDeduplication.getTimeCase match {
+              // Backward-compatibility, will not  be set for new entries
+              case TimeCase.DEDUPLICATED_UNTIL =>
+                Some(parseTimestamp(commandDeduplication.getDeduplicatedUntil))
+              // Set during normal execution, no time skews are added
+              case TimeCase.RECORD_TIME =>
+                val storedDuplicateRecordTime =
+                  parseTimestamp(commandDeduplication.getRecordTime)
+                Some(
+                  storedDuplicateRecordTime
+                    .add(commandDeduplicationDuration)
+                )
+              // Set during pre-execution, time skews are already accounted for
+              case TimeCase.MAX_RECORD_TIME =>
+                val maxRecordTime =
+                  parseTimestamp(commandDeduplication.getMaxRecordTime)
+                Some(
+                  maxRecordTime
+                    .add(commandDeduplicationDuration)
+                )
+              case TimeCase.TIME_NOT_SET =>
+                None
+            }
+          )
+          .forall(deduplicatedUntil =>
+            recordTimeOrMinimumRecordTime.isAfter(deduplicatedUntil.toInstant)
+          )
+        if (isDuplicate) {
+          StepContinue(transactionEntry)
+        } else {
+          if (commitContext.preExecute) {
+            // The out of time bounds entry is required in the committer, so we set it to the default value as we stop the steps here with the duplicate rejection
+            commitContext.outOfTimeBoundsLogEntry = Some(DamlLogEntry.getDefaultInstance)
           }
+          duplicateRejection(commitContext, transactionEntry)
         }
-        .getOrElse(StepContinue(transactionEntry))
+      }
+
+      private def duplicateRejection(
+          commitContext: CommitContext,
+          transactionEntry: DamlTransactionEntrySummary,
+      )(implicit loggingContext: LoggingContext) = {
+        rejections.reject(
+          DamlTransactionRejectionEntry.newBuilder
+            .setSubmitterInfo(transactionEntry.submitterInfo)
```

Comment:
I think you want to adjust the deduplication duration in the submitter info for the rejection for the skews. We don't know whether the earlier submission was actually committed at the maxRecordTime stored in command dedup entry; it could have been commited with any record time `t1` within the minSkew + maxSkew before the stored maxRecordTime. So if this rejection gets recorded at say `t2 = commitConntext.maxRecordTime`, we may have that `t2 - t1` > `deduplicationDuration` in `submitterInfo`. However, if you set the `deduplicationDuration` to `max(commitContext.maxRecordTime - maybeDedupValue.minRecordTime, submitterInfo.deduplicationDuration)`, then you're guaranteed to specify a duration during which a conflicting completion was committed. For that, you not only have to store the `max_record_time` in `DamlCommandDedupValue`, but also `min_record_time` because we don't know any more what the skew parameters were when the `DamlCommandDedupValue` was written.

Reply:
Done as port of [`4dd6e78` (#11634)](https://github.com/digital-asset/daml/pull/11634/commits/4dd6e784464ea56b12b53004a276dca12a768e30)